### PR TITLE
split services configuration in order to allow dynamic configuration

### PR DESCRIPTION
**Pull request by warlof**
split services accross multiple docker compose files allowing easier swap and dynamic configuration.

- docker-compose.mariadb.yml : deploy mariadb container which will be used as SeAT database
- docker-compose.proxy.yml : bind host port to front container port if SeAT should be deployed behind a proxying
- docker-compose.traefik.yml : deploy traefik container which will be used to redirect traffic against SeAT front container

main services (front, scheduler, cache and workers) will be stored in main docker-compose.yml

in order to publish SeAT using traefik and mariadb, run `docker compose -f docker-compose.yml -f docker-compose.mariadb.yml -f docker-compose.traefik.yml up`

in order to publish SeAT using mariadb and either serving it directly accross the web - or behind a proxy, run `docker compose -f docker-compose.yml -f docker-compose.mariadb.yml -f docker-compose.proxy.yml up`

We also add some toy for end user in order to highligh when SeAT is ready to handle request (something more visible than the apache output)

image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,12 +21,15 @@ RUN pecl install redis && \
...
 RUN cd /var/www && \
-    composer create-project eveseat/seat:5.0.x-dev --stability dev --no-scripts --no-dev --no-ansi --no-progress --prefer-source && \
+    composer create-project eveseat/seat:5.0.x-dev --stability dev --no-scripts --no-dev --no-ansi --no-progress && \
     composer clear-cache --no-ansi && \
     # Fix up the source permissions to be owned by www-data
-    chown -R www-data:www-data /var/www/seat && \
+    chown -R seat:seat /var/www/seat && \
     cd /var/www/seat && \
     # Setup the default configuration file
     php -r "file_exists('.env') || copy('.env.example', '.env');"
```
Comment (recursivetree) [inline Dockerfile:35]: What is this for? I

Comment (recursivetree) [reply Dockerfile:35]: Wait, it selected the wrong line. Apparently I can only comment on changes.

What I wanted to ask is why we have the .env setup. Isn't that provided from the outside?

Comment (warlof) [reply Dockerfile:35]: the `.env` inside `seat-docker` will be used as initial container seeding for environment variables and `docker-compose` files
this is the `.env` from the core - which might invole a little bit when the docker one will not - in such case, value are loaded this way :
1. config
2. environment variable
3. `.env` file

> lower is overriden by upper

**Comment by recursivetree** [comment]
Found an issue with the .env file: `APP_URL` is set after it used to define `EVE_CALLBACK_URL`. That gives me the following error: `WARN[0000] The "APP_URL" variable is not set. Defaulting to a blank string. `. It can be fixed by moving the  `APP_URL` definition up before the `EVE_CALLBACK_URL`.

My docker version is `Docker version 20.10.21, build baeda1f`

**Comment by warlof** [comment]
Nice catch, I was wondering why it was saying that while the value was set.

**Pull request merged**